rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -317,17 +597,25 @@ NodeParameters::describe_parameters(const std::vector<std::string> & names) cons
 {
   std::lock_guard<std::mutex> lock(mutex_);
   std::vector<rcl_interfaces::msg::ParameterDescriptor> results;
```

Comment:
@wjwwood nit: consider doing `results.reserve(names.size())` to avoid incremental reallocations below.